sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -409,9 +447,18 @@ private void initializeDatasetService(PipelineOptions pipelineOptions) {
       }
     }
 
+    @Setup
+    public void setup() {
```

Comment:
I'm not sure what this is adding?

Reply:
I've seen evidence that closing one stream append client for the `_default` stream causes a cascade close of the other ones, so moving the state for destinations to be reused between bundle executions at least decreased the occurrences of those cascading closes. 

I can revert this if this is not the right idea to try.